### PR TITLE
Current Docker documentation need fixes to work without error on Mabook M1

### PR DESCRIPTION
**Pull request by Aigerim-ai**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ashwin-pc** [comment]
@Aigerim-ai Thanks for the PR, can you add a description to the PR as to why this change is necessary? We also have another PR that wants to update the docs that cover mac setup. Want to make sure we understand both your issuesand how your solution fixed it.

Other PR for reference: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3620

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ To develop on OpenSearch Dashboards, you'll need:
 
 - A [GitHub account](https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account)
 - [`git`](https://git-scm.com/) for version control
-- [`Node.js`](https://nodejs.org/), [`npm`](https://www.npmjs.com/), and [`Yarn`](https://yarnpkg.com/) for building and running the project
+- [`Node.js`](https://nodejs.org/), [`npm`](https://www.npmjs.com/)( Note that, for Darwin (MacOS) only, it is important to have the latest version of Node.js in your machine, in other case you will have an issue with [`grunt`]plugin ), and [`Yarn`](https://yarnpkg.com/) for building and running the project
```
Comment (joshuarrrr) [inline DEVELOPER_GUIDE.md:46]: Are you saying here that you need a version _newer_ than the version in `.nvmrc`? Or just specifying that the version of node installed is important (which it is). See the note on L60 for the source of truth about the expected node version.

File: Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ RUN apt-get update && \
 # Specify the version of Chrome that matches the version of chromedriver in the package.json.
 # A list of Chrome versions can be found here:
 # https://www.ubuntuupdates.org/package/google_chrome/stable/main/base/google-chrome-stable
-ARG CHROME_VERSION=107.0.5304.121-1
+ARG CHROME_VERSION=107.0.3
```
Comment (joshuarrrr) [inline Dockerfile:16]: I don't believe this should be necessary - can you explain the purpose of rolling back the chrome version to an older one?

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (Aigerim-ai) [reply DEVELOPER_GUIDE.md:46]: Hi, thank you for your response, let me explain my situation

There was a problem when I followed our developer's guide. I had initially node installed in my MacOS. After installing nvm 14.20.1, I faced one problem, specifically package incompatibility(I will attach my screenshots below). 

Error: grunt@1.6.1. The engine node is incompatible with this module. Expected version >=16 Got 14.20.1. 

However, when I switched to the 16 th version of nvm, I had package incompatibility with opensearch-dashboard@3.0.0. So, this was very strange for me. How did I resolved this isssue?
I wrote on my laptop's terminal:
aigerimsuleimenova@Aigerims-MacBook-Air ~ % npm install n -g
added 1 package in 1s,
 Doing this before, I installed node by  npm install.
This command added one package, which other version on npm can not provide.

File: Dockerfile (hunk 2 above)
Comment (Aigerim-ai) [reply Dockerfile:16]: Yes please, I will  delete this

File: .yarnrc [hunk 3]
```diff
@@ -1 +1,2 @@
 --ignore-workspace-root-check true
+network-timeout 1000000
```
[the remark this answers is not in the document]
Comment (Aigerim-ai) [reply .yarnrc:2]: I will remove this line too

**Comment by ashwin-pc** [comment]
While i also ran into this problem, i dont think that this is the best way to solve the problem, especially if we are updating the project documentation.Considering that many others have faced the same issue and have raised Pr's to update the same content @Abyl10 can you add you issue and what are some of the steps you did to debug the issue here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3731. It will help us identify the common thread and update the docs better :)

**Comment by abbyhu2000** [comment]
Thank you @Aigerim-ai for your time on documenting the issue and opening the PR! Since we received multiple similar PRs on trouble shooting the docker manual guide, we created issue #3731 to collect feedback before moving forward. Closing this PR for now, can be reopened once we reach a conclusion.

**Pull request closed without merging**